nuimo/src/main/kotlin/com/senic/nuimo/NuimoBluetoothController.kt

```diff
@@ -175,9 +178,17 @@ class NuimoBluetoothController(bluetoothDevice: BluetoothDevice, context: Contex
                     val batteryPercentage = characteristic.getIntValue(BluetoothGattCharacteristic.FORMAT_UINT8, 0) ?: -1
                     notifyListeners { it.onBatteryPercentageChange(batteryPercentage) }
                 }
+                HARDWARE_REVISION_CHARACTERISTIC_UUID -> {
+                    hardwareVersion = characteristic.getStringValue(0)
+                    if (hardwareVersion?.length ?: 0 <= 2) hardwareVersion = null  // For old firmware we ignore the 19 string; It should be "MAJOR.MINOR"
+                }
                 FIRMWARE_VERSION_CHARACTERISTIC_UUID -> {
-                    val firmwareVersion = characteristic.getStringValue(0)
-                    if (firmwareVersion != null) notifyListeners { it.onFirmwareVersionRead(firmwareVersion) }
+                    firmwareVersion = characteristic.getStringValue(0)
+                }
+                MODEL_NUMBER_CHARACTERISTIC_UUID -> {
+                    color = characteristic.getStringValue(0)
+                    if (color?.length ?: 0 <= 1) color = null // For old firmware we ignore the 0 character; It should be a string of at least 2 characters
+                    notifyListeners { it.onInformationRead(hardwareVersion ?: "Unknown", firmwareVersion ?: "Unknown", color ?: "Unknown") }
```

Comment:
I understand that providing same values for older firmware versions is a good thing. We however assume that
a) users will update Nuimo's firmware
b) UI can still show "Unknown" if needed
c) Firmware version should either be a version, e.g. `2.2.3` or better not set at all.